doc/user/content/sql/explain.md

```diff
@@ -137,6 +137,25 @@ operator use the outputs of chains `%0`, `%1` and `%2` as its inputs.
 Many operators need to refer to columns in their input. These are displayed like
 `#3` for column number 3. (Columns are numbered starting from column 0).
 
+If you are explaining view creation for a view or one-off query that refers to unmaterialized sources, you may see `Filter` or `Project` operators attached to the unmaterialized source. This indicates that those operators will run before the upsert operator, and the upsert operator will not maintain state corresponding to data removed by the filter and project operators. For example:
+
+```
```

Comment:
I'm confused by what is being shown in the example. The example does not include any Filter or Project operators attached to an unmaterialized source.

A couple of clarifications on https://github.com/MaterializeInc/materialize/issues/8636#issuecomment-941635479:
1) a user will never see "(UID)" in `EXPLAIN` output; instead they will see "(u<some number>)".  It's just that in we replace "(u<some number>)" in tests so that we don't have to change a bunch of files if our user object naming scheme changes.
2) If a materialized view or query is referencing an unmaterialized source, Materialize will first read the source in once. Reading the source has the following steps:
```
(Pipe data in from source) --> 
(decode the row format) -->
(Filter/Project if they exist) -->
(Upsert if the envelope is Upsert)
```
So actually the filter + project will run as part of reading the source for all types of sources, not just upsert ones, but the filter + project will only noticeably affect the memory consumption for upsert sources.

So here's how I would interpret the plan in the comment:

```
> CREATE MATERIALIZED VIEW mv as SELECT a, c FROM data EXCEPT ALL SELECT a, c FROM data2 where d is null;

? EXPLAIN PLAN FOR VIEW mv;
Source materialize.public.data (u1): <-- unmaterialized source 1
| Project (#0, #2) <-- happens while reading the source

Source materialize.public.data2 (u3): <-- unmaterialized source 2
| Filter isnull(#3) <-- \
| Project (#0, #2, #3) <-- happen while reading the source

Query: <-- the materialized view
%0 =
| Get materialize.public.data (u1)
| Project (#0, #2) <-- no-op because it already ran as part of reading the "Source materialize.public.data (u1)"

%1 =
| Get materialize.public.data2 (u3)
| Filter isnull(#3) <-- \
| Project (#0, #2) <-- no-ops because they already ran as part of reading the "Source materialize.public.data2 (u3)"
| Negate

%2 =
| Union %0 %1
| Threshold
```